spring/boot-autoconfigure/src/test/java/com/linecorp/armeria/internal/spring/ArmeriaConfigurationUtilTest.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright 2018 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.internal.spring;
+
+import static com.linecorp.armeria.internal.spring.ArmeriaConfigurationUtil.configureAnnotatedHttpServices;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.reset;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.verify;
+
+import java.util.function.Function;
+
+import org.junit.Test;
+
+import com.google.common.collect.ImmutableList;
+
+import com.linecorp.armeria.common.HttpRequest;
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.common.HttpStatus;
+import com.linecorp.armeria.internal.annotation.AnnotatedHttpService;
+import com.linecorp.armeria.server.AbstractHttpService;
+import com.linecorp.armeria.server.Server;
+import com.linecorp.armeria.server.ServerBuilder;
+import com.linecorp.armeria.server.Service;
+import com.linecorp.armeria.server.ServiceRequestContext;
+import com.linecorp.armeria.server.annotation.Get;
+import com.linecorp.armeria.spring.AnnotatedServiceRegistrationBean;
+import com.linecorp.armeria.spring.MeterIdPrefixFunctionFactory;
+
+public class ArmeriaConfigurationUtilTest {
+
+    @Test
+    public void makesSureDecoratorsAreConfigured() {
+        final Function<Service<HttpRequest, HttpResponse>,
+                ? extends Service<HttpRequest, HttpResponse>> decorator = spy(new IdentityFunction());
+        final AnnotatedServiceRegistrationBean bean = new AnnotatedServiceRegistrationBean()
+                .setServiceName("test")
+                .setService(new SimpleService())
+                .setDecorators(decorator);
+
+        final ServerBuilder sb1 = new ServerBuilder();
+        configureAnnotatedHttpServices(sb1, ImmutableList.of(bean), MeterIdPrefixFunctionFactory.DEFAULT);
+        verify(decorator).apply(any());
+        assertThat(service(sb1)).isNotInstanceOf(AnnotatedHttpService.class)
+                                .satisfies(s -> assertThat(s.getClass().getName())
+                                        .endsWith("$ExceptionFilteredHttpResponseDecorator"));
```

Comment:
Could we:

- extract these three lines into a method to remove the duplicate assertion in `makesSureDecoratedServiceIsAdded()`?
- add a brief comment why an `AnnotatedHttpService` is decorated with `ExceptionFilteredHttpResponseDecorator` to the extracted method?
- add an extra assertion that checks if `DecoratingFunction` actually decorated `AnnotatedHttpService` to the extracted method? (To do this, you will have to define a subtype of `SimpleDecoratingService` instead of using an anonymous class.)
  - You can use `Service.as(Class)` method to check if a decorator exists in a service:
    ```java
    Service s = ...;
    assertThat(s.as(SomeDecoratorService.class)).isPresent();
    ```

Reply:
Thanks! I totally forgot the 3rd option! 😱